### PR TITLE
Fix a failure when reading data from a parquet file (and not a parquet directory)

### PR DESCRIPTION
**Pull request by selitvin**
Code was crashing since ParquetDataset's (pyarrow) partition_names attribute was set to None when the ParquetDataset object instance was created from a parquet file and not parquet directory.

### CODE REVIEW

File: petastorm/tests/test_parquet_reader.py [hunk 1]
```diff
@@ -70,6 +71,18 @@ def test_simple_read(scalar_dataset, reader_factory):
         _check_simple_reader(reader, scalar_dataset.data)
 
 
+@pytest.mark.parametrize('reader_factory', _D)
+def test_simple_read(scalar_dataset, reader_factory):
+    """See if we can read data when a single parquet file is specified instead of a parquet directory"""
+    assert scalar_dataset.url.startswith('file://')
```
Comment (selitvin) [inline petastorm/tests/test_parquet_reader.py:77]: Although our fix affects only make_reader(..) readers, let's check that the same issue does not occur with make_batch_reader.

Comment (chongxiaoc) [reply petastorm/tests/test_parquet_reader.py:77]: Unit test fails due to using same function name as line 68

**Review by chongxiaoc — CHANGES_REQUESTED** [review]
LGTM, just fix unit test name issue.

**Review by chongxiaoc — APPROVED** [review]
LGTM

**Pull request merged**